cmd/registry/cmd/upload/upload_test.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package upload
+
+import (
+	"testing"
+
+	"github.com/apigee/registry/server/registry"
+	"github.com/apigee/registry/server/registry/test/grpctest"
+)
+
+// TestMain will set up a local RegistryServer and grpc.Server for all
+// tests in this package if APG_REGISTRY_ADDRESS env var is not set
+// for the client.
+func TestMain(m *testing.M) {
```

Comment:
I'm guessing that this would need to be done for every package under `cmd`, right?

Reply:
That seems like a high line-count impact. Could this initialization be made implicit or moved up a level?

Reply:
I believe [TestMain()](https://pkg.go.dev/testing#hdr-Main) is the best / only official Go test mechanism to do this and it's package-specific.